spec/stack.test.js

```diff
@@ -16,9 +16,73 @@ describe('Stack', () => {
   context('push()', () => {
     it('pushes an element to the top of the stack.', () => {
       const myStack = new Stack()
+      const otherStack = new Stack('one')
+      const thirdStack = new Stack([{item11: 'item1'}, {item22: 'item2'}])
+      const quarterStack = new Stack({key1: 'value1'})
 
       expect(() => myStack.push('foo'))
         .to.alter(() => myStack.length(), { from: 0, to: 1 })
+      expect(myStack.items).to.eql(['foo'])
+
+      expect(() => otherStack.push(['bar', 'baz']))
+        .to.alter(() => otherStack.length(), { from: 1, to: 3 })
+      expect(otherStack.items).to.eql(['bar', 'baz', 'one'])
+
+      expect(() => thirdStack.push('hello'))
+        .to.alter(() => thirdStack.length(), { from: 2, to: 3 })
+      expect(thirdStack.items).to.eql([
+        'hello',
+        {item11: 'item1'},
+        {item22: 'item2'}
+      ])
+
+      expect(() => quarterStack.push(['world']))
+        .to.alter(() => quarterStack.length(), { from: 1, to: 2 })
+      expect(quarterStack.items).to.eql(['world', {key1: 'value1'}])
+    })
+  })
+
+  context('pop()', () => {
+    it('returns and removes the top element of the stack or null if the stack is empty', () => {
```

Comment:
The `or` implies there are two tests here:
1. Removes and returns
1. Returns null when empty